sisl/_category.py

```diff
@@ -196,7 +208,18 @@ def __invert__(self):
 
 
 @set_module("sisl.category")
-class NullCategory(Category):
+class GenericCategory(Category):
```

Comment:
I think we don't need this, we should simply register `Category` in `_sanitize_atoms` if the user passed categories which does not accept the geom as arguments errors will be raised.
This will allow us to enable `OrbitalCategory` and not need to register those as individuals.

Reply:
Hmm, I see your point, but I think for maintainability this is better, no?

Reply:
Yes, maybe it will look a bit ugly if you start to register lots of categories. Still, maybe it's good to keep the distinction even if we're not using it in this case for quick type checking (in sisl or user code)? 

PS: Does it make sense to do type checking in the category itself to raise a meaningful error or is it a waste of time that will decrease performance?

Reply:
> Yes, maybe it will look a bit ugly if you start to register lots of categories. Still, maybe it's good to keep the distinction even if we're not using it in this case for quick type checking (in sisl or user code)?

Hmm, perhaps the generic is a good idea in any case. Let me play a bit with `__instancecheck__` ;)

> PS: Does it make sense to do type checking in the category itself to raise a meaningful error or is it a waste of time that will decrease performance?
Bad idea, and yeah, for large systems this would be a performance hit...

Reply:
Sadly, that won't work... hmm...

Reply:
Having played a bit around with this. Yes, I think we should have the generic class. So keep it as is!